app/controllers/users_controller.rb

```diff
@@ -12,11 +12,13 @@ def index
 
   # GET /users/1
   def show
+    @search_terms = params["q"].presence
     @can_edit = can_edit?
     @my_dashboard = current_user.id == @user.id
-    @unfinished_works = Work.unfinished_works(@user)
-    @completed_works = Work.completed_works(@user)
-    @withdrawn_works = Work.withdrawn_works(@user)
+    @unfinished_works = Work.unfinished_works(@user, @search_terms)
+    @completed_works = Work.completed_works(@user, @search_terms)
+    @withdrawn_works = Work.withdrawn_works(@user, @search_terms)
+    @works_found = @unfinished_works.count + @completed_works.count + @withdrawn_works.count
```

Comment:
My understanding is that `count` always does a separate SQL `COUNT` query, while `size` is usually preferred because it will use `length` if the data is already in memory.

Reply:
Since this is an Array, I doubt it is sending a SQL query.    But prioritizing length over count seems like a reasonable thing.  
```
[4] pry(main)> unfinished.count
   (12.3ms)  SELECT COUNT(*) FROM "works"
=> 2
[5] pry(main)> unfinished.to_a.count
=> 2
[6] pry(main)> unfinished.length
=> 2
```

Reply:
Changed to length.